### PR TITLE
add forks api

### PR DESCRIPTION
**Pull request by terakilobyte**
Thanks for this library! I'd like to use it for a project I have in mind that involves using the [`forks`](https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#forks) API and thought I'd try adding it to octocrab.

I'm relatively new to Rust so if I've done something silly please don't hesitate to call it out, and if this is an unwelcome addition that's ok too.

On a side note, it appears dependabot updated tokio to 1.0, but reqwest is currently on 0.2. Unless you manually set it back, it results in a panic similar to [this](https://github.com/tokio-rs/tokio/issues/1837) when you run any example or use it in your own project and specify tokio 1.0 :frowning_face:

### CODE REVIEW

**Review by XAMPPRocky — CHANGES_REQUESTED** [review]
Thank you for your PR! Overall this looks good, I've left some comments of some minor changes, and once they've been implemented we can merge this.

> On a side note, it appears dependabot updated tokio to 1.0, but reqwest is currently on 0.2. Unless you manually set it back, it results in a panic similar to this when you run any example or use it in your own project and specify tokio 1.0 ☹️

That's unfortunate, I'll revert that change.

File: src/api/repos/forks.rs [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    /// # }
+    /// ```
+    /// Optionally, specify a sort order with [`ForkSort`].
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// use octocrab::repos::forks::ForkSort;
+    /// let forks = octocrab::instance().repos("owner", "repo").list_forks().sort(ForkSort::Oldest).send().await?;
+    /// # Ok(())
+    /// # }
+    /// ```
```
[the remark this answers is not in the document]
Comment (terakilobyte) [reply src/api/repos/forks.rs]: I'll give it a try.

File: src/api/repos/forks.rs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    pub fn list_forks(&self) -> ListForksBuilder<'_, '_> {
+        ListForksBuilder::new(self)
+    }
+
+    /// Creates a fork of a repository.
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// let new_fork = octocrab::instance().repos("owner", "repo").create_fork().send().await?;
```
[the remark this answers is not in the document]
Comment (terakilobyte) [reply src/api/repos/forks.rs]: I'll give it a try.

**Comment by terakilobyte** [comment]
I've modified the documentation, and added it into the entrance docs in a separate commit for easier removal if it's not wanted. 

One thing I noticed when updating the example and the documentation was that `Into<u8>` seems to infer the type correctly `per_page(30)`, but `Into<u32>` wasn't and I had to explicitly specify the type.

File: src/params.rs [hunk 3]
```diff
@@ -272,6 +272,18 @@ pub mod repos {
             f.write_str(&self.full_ref_url())
         }
     }
+    pub mod forks {
+        /// The available methods to sort repository forks by.
+        #[derive(Debug, Clone, Copy, serde::Serialize)]
+        #[serde(rename_all = "snake_case")]
+        #[non_exhaustive]
+
```
Comment (XAMPPRocky) [inline src/params.rs:280]:
```suggestion
```

File: src/api/repos/forks.rs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+impl<'octo> RepoHandler<'octo> {
+    /// List forks of a repository. Optionally, specify the
+    /// [sort](./forks/struct.ListForksBuilder.html#method.sort) order,
+    /// [page](./forks/struct.ListForksBuilder.html#method.page),
+    /// and items [per_page](./forks/struct.ListForksBuilder.html#method.per_page)
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// use octocrab::params::repos::forks::Sort;
+    /// let forks = octocrab::instance().repos("owner", "repo").list_forks().sort(Sort::Oldest).page(2u32).per_page(30).send().await?;
```
Comment (XAMPPRocky) [inline src/api/repos/forks.rs:94]:
```suggestion
    /// let forks = octocrab::instance()
    ///     .repos("owner", "repo")
    ///     .list_forks()
    ///     // Optional Parameters
    ///     .sort(Sort::Oldest)
    ///     .page(2u32)
    ///     .per_page(30)
    ///     .send()
    ///     .await?;
```

File: src/api/repos/forks.rs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    pub fn list_forks(&self) -> ListForksBuilder<'_, '_> {
+        ListForksBuilder::new(self)
+    }
+
+    /// Creates a fork of a repository. Optionally, specify the target
+    /// [organization](./forks/struct.CreateForkBuilder.html#method.organization) to
+    /// create the fork in.
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// let new_fork = octocrab::instance().repos("owner", "repo").create_fork().organization("weyland-yutani").send().await?;
```
Comment (XAMPPRocky) [inline src/api/repos/forks.rs:107]:
```suggestion
    /// let new_fork = octocrab::instance()
    ///     .repos("owner", "repo")
    ///     .create_fork()
    ///     // Optional Parameters
    ///     .organization("weyland-yutani")
    ///     .send()
    ///     .await?;
```

File: src/api/repos/forks.rs [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+            repo = self.handler.repo
+        );
+        self.handler.crab.post(url, Some(&self)).await
+    }
+}
+
+impl<'octo> RepoHandler<'octo> {
+    /// List forks of a repository. Optionally, specify the
+    /// [sort](./forks/struct.ListForksBuilder.html#method.sort) order,
+    /// [page](./forks/struct.ListForksBuilder.html#method.page),
```
Comment (XAMPPRocky) [inline src/api/repos/forks.rs:89]: Can you use [intra-doc](https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html) links to link to methods?

**Comment by XAMPPRocky** [comment]
>One thing I noticed when updating the example and the documentation was that Into<u8> seems to infer the type correctly per_page(30), but Into<u32> wasn't and I had to explicitly specify the type. I'm not sure why this is. This is a minimal reproduction in the rust playground.

Yeah that is a quirk of the language, specifying with a type suffix (e.g. `1u32`) is fine.

File: src/api/repos/forks.rs (hunk 6 above)
Comment (terakilobyte) [reply src/api/repos/forks.rs]: done

**Comment by XAMPPRocky** [comment]
Thank you again for your PR, and congrats on your first contribution! 🎉

**Comment by XAMPPRocky** [comment]
I've published this change as part of `0.8.14`

**Pull request merged**